app/client/keybase/keyvault.go

```diff
@@ -0,0 +1,296 @@
+	"encoding/json"
+	"errors"
+	"fmt"
+
+	"github.com/hashicorp/vault/api"
+	"github.com/pokt-network/pocket/shared/crypto"
+)
+
+// VaultKeybase implements the Keybase interface using HashiCorp Vault
+type VaultKeybase struct {
+	client *api.Client
+	mount  string
+}
+
+// NewVaultKeybase returns a new instance of VaultKeybase
+func NewVaultKeybase(client *api.Client, mount string) *VaultKeybase {
+	return &VaultKeybase{
+		client: client,
+		mount:  mount,
+	}
+}
+
+// Create new keypair entry in Vault
+func (vk *VaultKeybase) Create(passphrase, hint string) error {
+	keyPair, err := crypto.CreateNewKey(passphrase, hint)
+	if err != nil {
+		return err
+	}
+
+	keyPairBytes, err := json.Marshal(keyPair)
+	if err != nil {
+		return err
+	}
+
+	data := map[string]interface{}{
+		"key_pair": string(keyPairBytes),
+		"hint":     hint,
+	}
+
+	_, err = vk.client.KVv2(vk.mount).Put(context.TODO(), keyPair.GetAddressString(), data)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Import a new keypair from the private key hex string provided into Vault
+func (vk *VaultKeybase) ImportFromString(privStr, passphrase, hint string) error {
+	keyPair, err := crypto.CreateNewKeyFromString(privStr, passphrase, hint)
+	if err != nil {
+		return err
+	}
+
+	keyPairBytes, err := json.Marshal(keyPair)
+	if err != nil {
+		return err
+	}
+
+	data := map[string]interface{}{
+		"key_pair": string(keyPairBytes),
+		"hint":     hint,
+	}
+
+	_, err = vk.client.KVv2(vk.mount).Put(context.TODO(), keyPair.GetAddressString(), data)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Import a new keypair from the JSON string of the encrypted private key into Vault
+func (vk *VaultKeybase) ImportFromJSON(jsonStr, passphrase string) error {
+	keyPair, err := crypto.ImportKeyFromJSON(jsonStr, passphrase)
+	if err != nil {
+		return err
+	}
+
+	keyPairBytes, err := json.Marshal(keyPair)
+	if err != nil {
+		return err
+	}
+
+	data := map[string]interface{}{
+		"key_pair": string(keyPairBytes),
+	}
+
+	_, err = vk.client.KVv2(vk.mount).Put(context.TODO(), keyPair.GetAddressString(), data)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Get a keypair from Vault
+func (vk *VaultKeybase) Get(address string) (crypto.KeyPair, error) {
+	data, err := vk.client.KVv2(vk.mount).Get(context.TODO(), address)
+	if err != nil {
+		return nil, err
+	}
+
+	if data == nil {
+		return nil, errors.New("key not found")
+	}
+
+	keyPairStr, ok := data.Data["key_pair"].(string)
+	if !ok {
+		return nil, errors.New("invalid key pair")
+	}
+
+	var keyPairStruct crypto.EncKeyPair
+	err = json.Unmarshal([]byte(keyPairStr), &keyPairStruct)
+	if err != nil {
+		return nil, err
+	}
+
+	return &keyPairStruct, nil
+}
+
+// Get a public key from Vault
+func (vk *VaultKeybase) GetPubKey(address string) (crypto.PublicKey, error) {
+	keyPair, err := vk.Get(address)
+	if err != nil {
+		return nil, err
+	}
+
+	return keyPair.GetPublicKey(), nil
+}
+
+// Get a private key from Vault
```

Comment:
NIT (here and elsewhere): let's use proper godoc. We're not doing it everywhere in the codebase yet but the team discussed that we'll put more effort into it.
```suggestion
// GetPrivKey gets a private key from the vault
```